ProjFS.Mac/PrjFSKext/KauthHandler.cpp

```diff
@@ -501,6 +503,38 @@ static int HandleVnodeOperation(
                     goto CleanupAndReturn;
                 }
             }
+            
+            if (ActionBitIsSet(action, KAUTH_VNODE_WRITE_DATA))
```

Comment:
Probably hard to find the {}'s but this check is done regardless of `FileFlags_IsEmpty`.  If you view the full file it's probably easier to line the {}'s up.

We do in fact have a functional test covering the case too, WriteWithoutClose.

Reply:
>Probably hard to find the {}'s but this check is done regardless of FileFlags_IsEmpty. If you view the full file it's probably easier to line the {}'s up.

Thanks! Yep that was my mistake, I must've jumped braces when seeing where this lined up.

> We do in fact have a functional test covering the case too, WriteWithoutClose.

The test in `WriteWithoutClose` appears to be covering the scenario where we write to an empty placeholder without having read from it first.  Will we get one VnodeOp callback in that scenario or two?

Assuming it's just one callback (and `FileFlags_IsEmpty` is set), it would be good to add a test opens/reads the file before writing to it so that we can test the scenario where the empty flag is not set when `KAUTH_VNODE_WRITE_DATA` is set.

Reply:
@wilbaker As an FYI in the kext we do get a READ followed by WRITE for these operations
The open causes a READ and the write causes the WRITE.
So it does test the flow we want.

I can see how that can be confusing though.  I'll tidy the naming here and add an explicit read so there is no ambiguity.